File: lab2/source/minus.cpp
#include "operations.h"
#include "factory.h"

class Minus : public Operations{
public:
    void Operation(std::stack<int> &stack, std::istream& fin) override {
        Empty_check(stack);
        int previous_top = stack.top();
        stack.pop();
        Empty_check(stack);
        int res = stack.top() - previous_top;
        stack.pop();
        stack.push(res);
    }

};

namespace {
    Operations* createMinus() {
        return new Minus;
    }
    bool x = OperationsFactory::getInstance()->RegisterOperation(std::string("-"), createMinus);
}

File: jaba/lab1/src/main/java/org/example/Minus.java
package org.example;

import java.io.OutputStream;
import java.util.Scanner;
import java.util.Stack;

public class Minus extends Operations{
    @Override
    public void Operation(Stack<Integer> stack, Scanner scanner, OutputStream s){
        emptyCheck(stack);
        int previous_top = stack.pop();
        emptyCheck(stack);
        int res = stack.pop() - previous_top;
        stack.push(res);
    }
}
